ref: Improve JS SDK Component Tracking Documentation (#5619)
getsentry/sentry-docs: src/platforms/javascript/guides/vue/features/component-tracking.mdx

---
title: Track Vue Components

> description: "Learn how Sentry's Vue SDK allows you to monitor the rendering performance of your application and its components"

Reviewer: ```suggestion
description: "Learn how Sentry's Vue SDK allows you to monitor the rendering performance of your application and its components."
```

## Usage

<Alert>

To set up component tracking, you need to configure performance monitoring. For details on how to do this, check out our [Performance documentation](/platforms/javascript/guides/vue/performance/).

</Alert>

### Initial Application Load

By default, the Vue SDK tracks the rendering performance of your app (that is, its root component) on the initial page load. This operation is represented in the page load transaction by the **`ui.vue.render`** span.

### Child Components

You can also track your app's child components to get more details about the rendering process. This feature will create spans for each tracked component instance. The spans are called **`ui.vue.[hook]`** where `[hook]` is replaced by each tracked lifecycle stage. For example, the span representing the mount stage (the time between `beforeMount` and `mounted`) is called `ui.vue.mount`.

> To set it up, you only need to add a few options in your `Sentry.init` call:

Author: No, you only need to specify the `trackComponents` option. The other two are optional. I think we can do two things here:
1. add an "(optional)" to each soon to be h4 option heading (e.g. "`timeout` (optional)")
2. make this explicitly clear in the sentence you highlighted

@imatwawana do you have preferences on how to do this? We probably can also do both

> To set it up, you only need to add a few options in your `Sentry.init` call:

Author: Oh and yes, you're totally right, these should be h4 headings. Will change that!

> To set it up, you only need to add a few options in your `Sentry.init` call:

Reviewer: ```suggestion
To set it up, add, at minimum, the [`trackComponents`](#trackcomponents) in your `Sentry.init` call. Optionally, you can also add the [`hooks`](#hooks), and [`timeout`](#timeout) options.
```

### `trackComponents`

This is the main option that controls which child components should be tracked. Set it to `true` to track all of them or specify a list of individual components you want to track:

```javascript
Sentry.init({
  // ...
  trackComponents: true,
  // OR
  trackComponents: [
    "App",
    "RwvHeader",
    "RwvFooter",
    "RwvArticleList",
    "Pagination",
  ],
});
```

The default is `false`.

### `hooks`

Control which lifecycle hooks should be tracked. This is helpful if, for example, you want to know if some components are removed during the initial page load, in which case you can add a `destroy` hook to the default:

```javascript
Sentry.init({
  // ...
  trackComponents: true
  hooks: ["mount", "update", "destroy"],
});
```

The following hooks are available to track: `['activate', 'create', 'destroy', 'mount', 'update']`

> Note that when specifying `hooks`, we use the simple verb rather than `before` and `-ed` pairs. For example, `destroy` is correct while `beforeDestroy` and `destroyed` are incorrect.

Reviewer: ```suggestion
Note that when specifying `hooks`, we use the simple verb rather than `before` and `-ed` pairs. For example, `destroy` is correct, while `beforeDestroy` and `destroyed` are incorrect.
```
